tests/test_dictionary.cpp

```diff
@@ -0,0 +1,149 @@
+/*************************************************************************/
+/*  test_dictionary.cpp                                                  */
+/*************************************************************************/
+/*                       This file is part of:                           */
+/*                           GODOT ENGINE                                */
+/*                      https://godotengine.org                          */
+/*************************************************************************/
+/* Copyright (c) 2007-2020 Juan Linietsky, Ariel Manzur.                 */
+/* Copyright (c) 2014-2020 Godot Engine contributors (cf. AUTHORS.md).   */
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "test_dictionary.h"
+#include "core/templates/ordered_hash_map.h"
+#include "core/templates/safe_refcount.h"
+#include "core/variant/dictionary.h"
+#include "core/variant/variant.h"
+#include "tests/test_macros.h"
+
+namespace TestDictionary {
+
+MainLoop *test() {
+	// test get_key_list.
+	{
+		Dictionary map;
+		List<Variant> keys;
+		auto ptr = &keys;
+		map.get_key_list(ptr);
+		CHECK(keys.empty());
+		map[1] = 3;
+		map.get_key_list(ptr);
+		CHECK(keys.size() == 1);
+		CHECK(int(keys[0]) == 1);
+		map[2] = 4;
+		map.get_key_list(ptr);
+		CHECK(keys.size() == 3);
+	}
+	// test get_key_at_index
+	{
+		Dictionary map;
+		map[4] = 3;
+		auto val = map.get_key_at_index(0);
+		CHECK(int(val) == 4);
```

Comment:
`auto` keyword is banned for C++ development in Godot. In this case, I think not using `auto` improves readability:

```suggestion
		auto val = map.get_key_at_index(0);
		CHECK(int(val) == 4);
```
```suggestion
		int val = map.get_key_at_index(0);
		CHECK(val == 4);
```

Reply:
I did have a question @Xrayez . Since a lot of the functions return Variant objects, where would I convert it to the actual object to improve readablity?

Reply:
With the changes I suggested, it should be possible to assign those values, the overloaded `Variant::operator=()` does conversion to any `Variant`-compatible datatype already. 🙂